### PR TITLE
Closes #50 Increase unit test coverage for derive_params_growth_height

### LINKED ISSUE #50: Documentation:  Increase unit-test coverage for `derive_params_growth_height()`
Create unit tests for `derive_params_growth_height()` using the pre-existing WHO metadata, be sure to include:

- edge cases such as extremely low or high height/lenghts
- subject that is under and over 2 years old between visits
- missing height/lengths, insertion of values like NA
- regular/as expected height/lengths

### PR DESCRIPTION
**Pull request by ahasoplakus**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralpeds (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralpeds site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by rossfarrugia** [comment]
@zdz2101 best for you to review this one please and double check against all David advised over email.

**Comment by Fanny-Gautier** [comment]
Is there any test for outlier data ?

**Comment by rossfarrugia** [comment]
> Is there any test for outlier data ?

@ahasoplakus

File: tests/testthat/test-derive_params_growth_height.R [hunk 1]
```diff
@@ -39,3 +39,144 @@ test_that("derive_params_growth_heightlength Test 1: derive_params_growth_height
     filter(actual, PARAMCD == "WT2HTZ") %>% pull(AVAL)
   )
 })
+
+## Test 2: derive_params_growth_heightlength works for subjects that are under or over 2 years old
+## between visits
+test_that("derive_params_growth_heightlength Test 2: derive_params_growth_height works for subjects that are under or over 2 years old between visits", { # nolint
+  vs_data <- tibble::tribble(
+    ~USUBJID, ~SEX, ~HEIGHT, ~HEIGHTU, ~VSTESTCD, ~VSSTRESN, ~AAGECUR,
```
Comment (zdz2101) [inline tests/testthat/test-derive_params_growth_height.R:47]: Don't know if you meant to separate the over/under by gender as well, but it would be nice to maybe do 1 each?

**Review by zdz2101 — CHANGES_REQUESTED** [review]
@ahasoplakus Thank you for all your help so far! Looks great, just needs a couple more examples, as cited above, is it possible to add outlier cases? Something like Z-scores of > |10| so both -10 and 10, with the currently pre-existing metadata, to make it easier you can generate fake data and use just one metadataset like the `who_wt_for_lgth_boys` to trim down the preprocessing work

**Comment by ahasoplakus** [comment]
> @ahasoplakus Thank you for all your help so far! Looks great, just needs a couple more examples, as cited above, is it possible to add outlier cases? Something like Z-scores of > |10| so both -10 and 10, with the currently pre-existing metadata, to make it easier you can generate fake data and use just one metadataset like the `who_wt_for_lgth_boys` to trim down the preprocessing work

Thanks @zdz2101 @Fanny-Gautier for your comments. Will add the outlier cases as suggested.

**Comment by rossfarrugia** [comment]
@ahasoplakus might be worth using {covr} to check any lines in this function that are untested by your PR, as it could help find one or two more useful tests to add. i'm hoping to see after this PR and Laura's for the other function that we'll get our code coverage score up above 90%, as that's then in line with other admiral packages.

**Comment by ahasoplakus** [comment]
> @ahasoplakus might be worth using {covr} to check any lines in this function that are untested by your PR, as it could help find one or two more useful tests to add. i'm hoping to see after this PR and Laura's for the other function that we'll get our code coverage score up above 90%, as that's then in line with other admiral packages.

@rossfarrugia I have ran `devtools::test_coverage_active_file()` which uses `file_coverage()` from {covr} and it seems all lines are covered
image

**Review by Fanny-Gautier — APPROVED** [review]
No further comments

**Comment by rossfarrugia** [comment]
Thanks @Fanny-Gautier! @zdz2101 please double check and approve if OK from your side.

File: tests/testthat/test-derive_params_growth_height.R [hunk 2]
```diff
@@ -39,3 +39,188 @@ test_that("derive_params_growth_heightlength Test 1: derive_params_growth_height
...
+  fake_meta_over <- tibble::tribble(
+    ~SEX, ~HEIGHT_LENGTH, ~HEIGHT_LENGTHU, ~L, ~M, ~S,
+    "M", 25, "cm", 7, 8, 9,
+    "M", 26, "cm", 10, 11, 12,
+    "F", 26, "cm", 13, 14, 15,
+    "F", 27, "cm", 15, 16, 17
+  )
+
+  actual_sds_under <- derive_params_growth_height(
+    dataset = filter(vs_data, AAGECUR < 730),
```
Comment (Fanny-Gautier) [inline tests/testthat/test-derive_params_growth_height.R:71]: Should the new argument `by_vars` be added there ?

File: tests/testthat/test-derive_params_growth_height.R [hunk 3]
```diff
@@ -39,3 +39,188 @@ test_that("derive_params_growth_heightlength Test 1: derive_params_growth_height
...
+    meta_criteria = fake_meta_under,
+    parameter = VSTESTCD == "WEIGHT",
+    analysis_var = VSSTRESN,
+    set_values_to_sds = exprs(
+      PARAMCD = "WT2HTZ"
+    )
+  )
+
+  actual_pctl_over <- derive_params_growth_height(
+    dataset = filter(vs_data, AAGECUR >= 730.5),
```
Comment (Fanny-Gautier) [inline tests/testthat/test-derive_params_growth_height.R:84]: Should the new argument `by_vars` be added there ?

File: tests/testthat/test-derive_params_growth_height.R [hunk 4]
```diff
@@ -39,3 +39,188 @@ test_that("derive_params_growth_heightlength Test 1: derive_params_growth_height
...
+  fake_meta <- tibble::tribble(
+    ~SEX, ~HEIGHT_LENGTH, ~HEIGHT_LENGTHU, ~L, ~M, ~S,
+    "M", 30, "cm", 1, 2, 3,
+    "M", 31, "cm", 4, 5, 6,
+    "F", 25, "cm", 7, 8, 9,
+    "F", 26, "cm", 10, 11, 12
+  )
+
+  out <- derive_params_growth_height(
+    dataset = vs_data,
```
Comment (Fanny-Gautier) [inline tests/testthat/test-derive_params_growth_height.R:177]: Should the new argument `by_vars` be added there ?

File: tests/testthat/test-derive_params_growth_height.R [hunk 5]
```diff
@@ -39,3 +39,188 @@ test_that("derive_params_growth_heightlength Test 1: derive_params_growth_height
...
+
+  fake_meta <- tibble::tribble(
+    ~SEX, ~HEIGHT_LENGTH, ~HEIGHT_LENGTHU, ~L, ~M, ~S,
+    "M", 30, "cm", 1, 2, 3,
+    "M", 31, "cm", 4, 5, 6
+  )
+
+  expect_error(
+    derive_params_growth_height(
+      dataset = vs_data,
```
Comment (Fanny-Gautier) [inline tests/testthat/test-derive_params_growth_height.R:216]: Should the new argument `by_vars` be added there ?

**Review by zdz2101 — APPROVED** [review]
